client/app.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react'
+import TodoForm from './todo-form.js'
+import TodoList from './todo-list.js'
+
+export default class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { todos: [] }
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  async handleSubmit(event) {
+    event.preventDefault()
+    const formData = new FormData(event.target)
+    const data = {
+      task: formData.get('taskInput'),
+      dueDate: formData.get('dueDateInput')
+    }
+    event.target.reset()
+    await fetch('/api/todos/', {
+      method: 'POST',
+      body: JSON.stringify(data),
+      headers: {
+        'content-type': 'application/json'
+      }
+    })
+
+    const res = await fetch('/api/todos')
```

Comment:
Instead of re-fetching all of the todos, just add the saved one to current `state`.